### ISSUE TITLE
Websocket: Origin Not Allowed

### ISSUE THREAD

**Issue by princessGasmask** [bakape/meguca#385]
Since I updated to 2.3, 2.2 won't even run for me anymore, and closes automatically, even with Rethink running. Anyways, on 2.3, when I try to connect via localhost:8000, I get the previous issue that I'd mentioned. With 127.0.0.1:8000, however, I get a blank version of meguka with the default theme. It also switches between "connecting" and "disconnected" states automatically, which made me look at meguca's output. This is what I get : 
![capture](https://cloud.githubusercontent.com/assets/5853862/19216683/140af3f0-8d8c-11e6-8470-e987bd5fe30f.PNG)

**Comment by bakape** [comment]
Since you are not serving from localhost, you have to specify that you are serving behind a reverse proxy.
`./meguca --reverse-proxied` or similar.

**Comment by princessGasmask** [comment]
That's the thing. I am serving from localhost.

**Comment by bakape** [comment]
Assuming your host file is fucked somehow 127.0.0.1 and localhost are not the same thing.

**Comment by princessGasmask** [comment]
Oh, okay. I understand now. So would the command for meguca be meguca
-reverse-proxied 127.0.0.1? Also, should I be reconfiguring rethinkdb in
any way? The fact that it all worked before and went to shit so quickly has
me second-guessing myself.

**Comment by bakape** [comment]
Just `meguca --reverse-proxied`. 

>Also, should I be reconfiguring rethinkdb in
any way?

I don't see why you would need to.

**Comment by princessGasmask** [comment]
Nothing has changed. Get the same websocket error.

**Comment by bakape** [comment]
It is likely an issue with your hosts file then. Normally localhost and
127.0.0.1 are aliases, so this should not be even happening. Try resetting
it.

**Comment by sserenade** [comment]
I am also having this issue - it's a pretty fresh install of Debian 8 with no modifications to the hosts file. Using the new binary 2.4.1 with nginx and modeled my conf around the one in the docs folder. nginx proxies the request no problem to :8000, but then I get `websocket: origin not allowed`. I have tried the `-reverse-proxied` flag as well as specifying the origin with `-origin` and am stumped.

**Comment by bakape** [comment]
Okay, maybe that is a bug then or something I missed in the docs. Can you post your nginx conf file, the line you use to start up meguca and the URL you use to access it?

**Comment by sserenade** [comment]
[nginx-site-conf.txt](https://github.com/bakape/meguca/files/545965/nginx-site-conf.txt)

I figured I'd rule everything out so I used the same setup as much as possible, letsencrypt certs and ssl.

I run meguca with `./meguca debug -reverse-proxied` . I've tried many variants with the ssl flag, and specifying origin as well. Something I did notice - if I run it with the debug flag, it shows 127.0.0.1, but without the flag it intersperses the client's actual ip in the output.

I'm trying to serve from [https://boards.sleeping.pizza/](url) .
